rostime/src/time.cpp

```diff
@@ -108,11 +108,15 @@ namespace ros
 #if HAS_CLOCK_GETTIME
     timespec start;
     clock_gettime(CLOCK_REALTIME, &start);
+    if (start.tv_sec < 0 || start.tv_sec > UINT_MAX)
```

Comment:
Same as the other pr, might want to use `UINT32_MAX`:

http://en.cppreference.com/w/cpp/types/integer